discord/guild_scheduled_event.go

```diff
@@ -35,21 +36,22 @@ type GuildScheduledEventCreate struct {
 	Name               string                     `json:"name"`
 	PrivacyLevel       ScheduledEventPrivacyLevel `json:"privacy_level"`
 	ScheduledStartTime time.Time                  `json:"scheduled_start_time"`
-	ScheduledEndTime   time.Time                  `json:"scheduled_end_time,omitempty"`
+	ScheduledEndTime   *time.Time                 `json:"scheduled_end_time,omitempty"`
 	Description        string                     `json:"description,omitempty"`
 	EntityType         ScheduledEventEntityType   `json:"entity_type"`
+	Image              *Icon                      `json:"image,omitempty"`
 }
 
 type GuildScheduledEventUpdate struct {
-	ChannelID          *snowflake.ID               `json:"channel_id,omitempty"`
-	EntityMetaData     *EntityMetaData             `json:"entity_metadata,omitempty"`
-	Name               string                      `json:"name,omitempty"`
-	PrivacyLevel       *ScheduledEventPrivacyLevel `json:"privacy_level,omitempty"`
-	ScheduledStartTime *time.Time                  `json:"scheduled_start_time,omitempty"`
-	ScheduledEndTime   *time.Time                  `json:"scheduled_end_time,omitempty"`
-	Description        *string                     `json:"description,omitempty"`
-	EntityType         *ScheduledEventEntityType   `json:"entity_type,omitempty"`
-	Status             *ScheduledEventStatus       `json:"status,omitempty"`
+	ChannelID          *snowflake.ID                  `json:"channel_id,omitempty"`
+	EntityMetaData     *json.Nullable[EntityMetaData] `json:"entity_metadata,omitempty"`
```

Comment:
I don't think this needs to be a `json.Nullable[T]` as the fields inside are the important part here. You can reset the fields inside by making them empty